sdk/servicebus/azure_messaging_servicebus/src/error.rs

```diff
@@ -123,12 +127,65 @@ impl From<azure_core::error::Error> for ServiceBusError {
             _ => ErrorKind::Unknown,
         };
 
-        ServiceBusError::new(kind, error.to_string())
+        ServiceBusError::with_source(kind, error.to_string(), error)
     }
 }
 
 impl From<azure_core_amqp::AmqpError> for ServiceBusError {
     fn from(error: azure_core_amqp::AmqpError) -> Self {
-        ServiceBusError::new(ErrorKind::Amqp, error.to_string())
+        ServiceBusError::with_source(ErrorKind::Amqp, error.to_string(), error)
```

Comment:
Do you expect to provide descriptive texts for servicebus errors generated from other errors? If your general practice is to just use `.to_string()`, there's not a ton of value in the `message` parameter.